src/gluonnlp/data/info.py

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""API to get list of datasets"""
+__all__ = ['list_datasets']
+
+from . import (conll, question_answering, sentiment,
+               super_glue, translation, word_embedding_evaluation,
+               intent_slot, glue, baidu_ernie_data)
+from .corpora import google_billion_word
+from .corpora import large_text_compression_benchmark
+from .corpora import wikitext
+
+
+def list_datasets():
+    """Returns the list of datasets
+    """
+    datasets = (conll.__all__ + glue.__all__ + sentiment.__all__ +
+                google_billion_word.__all__ + intent_slot.__all__ +
+                large_text_compression_benchmark.__all__ +
+                question_answering.__all__ + super_glue.__all__ +
+                translation.__all__ + word_embedding_evaluation.__all__ +
+                wikitext.__all__ + baidu_ernie_data.__all__)
```

Comment:
This function clashes with our dataset registry, which already provides `list_datasets`:

https://github.com/dmlc/gluon-nlp/blob/6fb43ec2cd1a90ade1a7fb545570c85270621134/src/gluonnlp/data/registry.py#L116-L144

The existing `list_datasests` requires that all datasets are registered with `@register`.
We don't have any mechanism to ensure new datasests are properly registered.

On the other hand, we don't have any mechanisms to ensure that `__all__` does not list anything else besides datasets.

For consistencies sake, the same approach should be used for registering & listing models, as is used for datasets.

CC: @eric-haibin-lin

Reply:
@leezu @eric-haibin-lin will changing the name of list_datasets() help resolve the conflict??

Reply:
The existing `list_datasets()` should already do the same thing as your `list_datasets()`, so changing the name will lead to having two more or less equivalent functions.

Reply:
what about list_models() ??

Reply:
`list_models` doesn't exist yet, so it would be great to add it.
It may make sense to follow the same "registry" approach for `list_models` instead of assuming that all types in `__all__` are models. What do you think?

Reply:
@eric-haibin-lin and @leezu  your thoughts on this so that we can proceed further.

Reply:
@leezu should I submit only with list_models() then??

Reply:
@avinashsai yes, you can delete the `list_datasets` from this PR